src/DependencyInjection/Compiler/RegisterHooksPass.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Find hook services and store them in an parameter.
+ *
+ * @author David Molineus <https://github.com/dmolineus>
+ */
+class RegisterHooksPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $serviceIds = $container->findTaggedServiceIds('contao.hook');
+        $hooks      = [];
+
+        foreach ($serviceIds as $serviceId => $tags) {
+            foreach ($tags as $attributes) {
+                $this->guardRequiredAttributesExist($serviceId, $attributes);
+
+                $priority = $attributes['priority'] ?? 0;
```

Comment:
You should probably int-cast this for safety reasons.

Reply:
Done in ec49f63.